src/main/java/Main.java

```diff
@@ -166,12 +150,44 @@ public static void main(String[] args) throws Exception {
 
             _main(args);
         } catch (UnsupportedClassVersionError e) {
-            System.err.println("Jenkins requires Java 8, but you are running "+
-                System.getProperty("java.runtime.version")+" from "+System.getProperty("java.home"));
+            System.err.println(String.format(
+                    "Jenkins requires Java versions %s but you are running with Java %s from %s",
+                    SUPPORTED_JAVA_VERSIONS, System.getProperty("java.specification.version"), System.getProperty("java.home"))
+            );
             e.printStackTrace();
         }
     }
 
+    /*package*/ static void verifyJavaVersion(int javaClassVersion, boolean enableFutureJava)
+            throws Error {
+        final String displayVersion = String.format("%d.0", javaClassVersion);
+        if (SUPPORTED_JAVA_CLASS_VERSIONS.contains(javaClassVersion)) {
+            // Fine
+        } else if (javaClassVersion > MINIMUM_JAVA_CLASS_VERSION) {
+            if (enableFutureJava) {
+                LOGGER.log(Level.WARNING,
```

Comment:
As @batmat asked me before, it's better to use the form: `log(Level level, String msg, Object params[]) ` avoiding the string format when the log level is under the one used. Just spreading your best practices :wink:

Reply:
Be sure I know about this best practice. I just copypasted the log entry from below. In this case it does not really matter, because the method is invoked only once on Jenkins startup. Also, we expect WARNING messages to be  printed on almost all Jenkins instances => ok